types/animation.d.ts

```diff
@@ -0,0 +1,43 @@
+import { PlayerObject } from './model'
+
+type AnimationFn = (player: PlayerObject, time: number) => void
+interface IAnimation {
+  play(player: PlayerObject, time: number): void
+}
+export type Animation =
+  | AnimationFn
+  | IAnimation
+  | {
```

Comment:
I don't think the definition is necessary. `AnimationFn` and `IAnimation` are enough. (this is definition is already included in `IAnimation`)